internal/app/init/pkg/system/services/ntpd.go

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// nolint: dupl,golint
+package services
+
+import (
+	"github.com/autonomy/talos/internal/app/init/pkg/ntpd"
+	"github.com/autonomy/talos/internal/app/init/pkg/system/conditions"
+	"github.com/autonomy/talos/internal/pkg/userdata"
+)
+
+// NTPd implements the Service interface. It serves as the concrete type with
+// the required methods.
+type NTPd struct {
+	client *ntpd.NTP
+}
+
+// ID implements the Service interface.
+func (t *NTPd) ID(data *userdata.UserData) string {
+	return "ntpd"
+}
+
+// PreFunc implements the Service interface.
+func (t *NTPd) PreFunc(data *userdata.UserData) error {
+	n := ntpd.New(data.Services.NTPd)
+	if err := n.Query(); err != nil {
+		return err
+	}
+	if err := n.SetTime(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// PostFunc implements the Service interface.
+func (t *NTPd) PostFunc(data *userdata.UserData) (err error) {
+	return nil
+}
+
+// ConditionFunc implements the Service interface.
+func (t *NTPd) ConditionFunc(data *userdata.UserData) conditions.ConditionFunc {
+	// what is this?
+	return conditions.None()
+}
+
+func (t *NTPd) Start(data *userdata.UserData) error {
+	go t.client.Daemon()
```

Comment:
Hmm, this is a different way than we are currently running other services. Not that I'm opposed to this but I'm just thinking out loud. The way we did it with `udevd` was to compile `udevd` as a binary and then use the on host "runner": https://github.com/autonomy/talos/blob/master/internal/app/init/pkg/system/services/udevd.go#L40-L60

I'm not opposed to running that as a baked in service (not a container, and not a separate process), but I'm curious if we want to be consistent. Either that or we create the notion of a service that is not a container or another process. Only reason I would want to aim for container or on host process is because when we can see it with `osctl ps` (currently only true for container based service, but we should extend to on host services in the future).